lib/linters/enhanced_idv_events_linter.rb

```diff
@@ -0,0 +1,220 @@
+module RuboCop
+  module Cop
+    module IdentityIdp
+      class EnhancedIdvEventsLinter < RuboCop::Cop::Base
+        extend AutoCorrector
+        include MultilineElementLineBreaks
+
+        RESTRICT_ON_SEND = [:track_event]
+
+        ENHANCED_ARGS = [:proofing_components].freeze
+
+        def on_send(track_event_send)
+          method = track_event_send.each_ancestor(:def).first
+          return if !should_check_method?(method.method_name)
+
+          ENHANCED_ARGS.each do |arg_name|
+            check_arg_present_on_event_method(arg_name, method)
+            check_arg_present_in_track_event_send(arg_name, track_event_send)
+            check_arg_has_docs(arg_name, method)
+          end
+        end
+
+        private
+
+        def check_arg_has_docs(arg_name, method)
+          pattern = Regexp.new("# @param \\[.+\\] #{arg_name}")
+          has_docs = preceding_lines(method).any? do |line|
+            line.is_a?(Parser::Source::Comment) && pattern.match?(line.text)
+          end
+
+          return if has_docs
+
+          add_offense(
+            method,
+            message: "Missing @param documentation comment for #{arg_name}",
+          ) do |corrector|
+            last_param_line = preceding_lines(method).reverse.find do |line|
+              line.is_a?(Parser::Source::Comment) && /@param/.match?(line.text)
+            end
+
+            comment = "# @param [Object] #{arg_name} TODO: Write doc comment"
```

Comment:
I wanted this autocorrect (adding doc comment) in place so I could get all the missing `@param` declarations in analytics_events.rb but would be happy to remove it if we want potentially avoid ending up with a mess of autogenerated "TODO" comments in the future

Reply:
(if say, a developer had a git precommit hook that automatically ran `rubocop -a` and committed the results)